Libraries/Components/CheckBox/AndroidCheckBoxNativeComponent.js

```diff
@@ -42,6 +42,7 @@ type NativeProps = $ReadOnly<{|
 
   on?: ?boolean,
   enabled?: boolean,
+  tintColors: {|true: ?number, false: ?number|} | typeof undefined,
```

Comment:
maybe this should be named `tintColors` so it follows the pattern of [switch](https://facebook.github.io/react-native/docs/switch#trackcolor

Reply:
Did you mean `trackColors`? I think `tintColors` is fine.

Reply:
oh, sorry, I meant `tintColor` - just make it singular. Plural makes a little more sense imho, but switch has it in singular...

Reply:
I think `tintColors` is fine. Not sure yet about moving Checkbox out of RN into a separate package.

Reply:
@janicduplessis I believe the type is correct here: 

> `tintColors?: ?{|true: ?number, false: ?number|}` so it is optional and can accept null and undefined

this is the type for the native component, and should correspond to what the [`getTintColors`](https://github.com/facebook/react-native/pull/18300/files#diff-7f3d990a8a1579593beee654dfc118feR161) function returns - which is either an object which always has two entries (where the values are either a number or undefined), or undefined, but not null

Reply:
I feel like passing an object with only true or false should work if you want to change the color of only one state. `{true: 'red'}` looks better than `{true: 'red', false: undefined}`. Afaik it should still work if we just change the flow type since native wont make a difference between the key not being there and having an undefined value.

https://flow.org/try/#0C4TwDgpgBAKglgO2AYQPYBtUCcDOUC8UA3gD7BYCuEAXFAPw7mIDmANFAGYCG6ON9jLCxIBfAFBiA9JKgBRLFjEBjVAkZRgiYLXhI0mXAWLkqtAORYIAEzMiA3BOlQA6tgDWOZavWakAJh0tfWw8QiITfgtrM3ZuXn4KBCsIDkRre0cZVywPLzVgDS0AZkC9DBCjcMpIyxtYnj5aBAp0dHsgA

Reply:
If it doesn't work for some reason then I'm fine with leaving the type as is.

Reply:
@janicduplessis I'm not sure we're both talking about the same thing - as a RN user, you can ofc pass just one color. 

From user perspective, the type of the prop is 

`tintColors?: {|true?: ColorValue, false?: ColorValue|}` 

this gets transformed to 

`tintColors: {|true: ?number, false: ?number|} | typeof undefined` 

which gets sent to native. Does this make sense?

We could change `tintColors?: {|true?: ColorValue, false?: ColorValue|}` to `tintColors?: {|true?: ?ColorValue, false?: ?ColorValue|}`, is that what you're after?

Reply:
I see, is there a reason for having different types for native prop `tintColors` and the component prop `tintColors`? I think using the more permissive type `tintColors?: ?{|true?: ?number, false?: ?number|}` (all props are optional and nullable) for both would be better.

Reply:
Ok I've re-read your initial comment and it makes sense now. I still think using the same type is slightly clearer but either way works.

Reply:
> is there a reason for having different types for native prop `tintColors` and the component prop `tintColors`

yes - the android system works with colors that are `int`s and react-native works with string colors. The conversion is done by [`processColor`](https://github.com/facebook/react-native/blob/master/Libraries/StyleSheet/processColor.js)

Do you want me to change `tintColors?: {|true?: ColorValue, false?: ColorValue|}` to `tintColors?: {|true?: ?ColorValue, false?: ?ColorValue|}` which is a little more permissive?

Reply:
Makes sense, yea let’s do this then this is good to ship!